NitroxClient/Communication/Packets/Processors/VehicleColorChangeProcessor.cs

```diff
@@ -31,36 +30,29 @@ public override void Process(VehicleColorChange colorPacket)
                 GameObject baseCell = moonpool.RequireComponentInParent<BaseCell>().gameObject;
 
                 subNameInput = baseCell.RequireComponentInChildren<SubNameInput>();
-                subNameTarget = subNameInput.target;
             }
 
             else
             {
                 subNameInput = vehicleObject.RequireComponentInChildren<SubNameInput>();
-                subNameTarget = subNameInput.target;
             }
 
-            if (subNameInput && subNameTarget)
+            using (packetSender.Suppress<VehicleColorChange>())
             {
-                if (vehicleObject.GetComponent<Vehicle>())
+                if (subNameInput && subNameInput.target)
                 {
                     // Switch to the currently selected tab
                     subNameInput.SetSelected(colorPacket.Index);
 
                     // OnColorChange calls these two methods, in order to update the vehicle color and the color+text on the ingame panel, respectively:
-                    subNameTarget.SetColor(colorPacket.Index, colorPacket.HSB.ToUnity(), colorPacket.Color.ToUnity());
+                    subNameInput.target.SetColor(colorPacket.Index, colorPacket.HSB.ToUnity(), colorPacket.Color.ToUnity());
                     subNameInput.SetColor(colorPacket.Index, colorPacket.Color.ToUnity());
-
                 }
                 else
                 {
-                    Log.Error($"[VehicleColorChangeProcessor] The GameObject {vehicleObject.name} doesn't have a Vehicle component");
+                    Log.Error($"[VehicleColorChangeProcessor] SubNameInput or targeted SubName is null.");
```

Comment:
When a player is still on the console and another one leaves the moonpool with the seamoth (for example), the first player still sees the upgrade console available with the color changing and name changing panel. While it should go back to the "Nothing docked" panel. This error will then spam in the logs when changing the color / name (It's the same for the VehicleNameChangeProcessor)

Reply:
Oh, didn't tried to undock the seamoth while testing :open_mouth: 
I will have to check this.

Reply:
Isn't that an issue of the undock processor then? Maybe the best place to look at

Reply:
well, I don't know. Should I fix it too in this PR or make a new PR for this (me or someone else) ? :thinking:

Reply:
> well, I don't know. Should I fix it too in this PR or make a new PR for this (me or someone else) ? :thinking: 

If you find a fix before it's merged by me/other Gost you can include it here. If not it's not that problematic if you have a null check in the patch to stop the packet spamming even if the ui is "broken" then.